design/one-pager-component-mutable-and-versioning.md

```diff
@@ -183,13 +184,24 @@ metadata:
   name: fancytraits.core.oam.dev
 spec:
   revision:
-    aware: true
+    alwaysControl: true
     revisonPath:
       - "spec.revision"
   definitionRef:
     name: fancytraits.core.oam.dev
 ```
 
+#### Revision field rule
+
+Versioning mechanism will be related with `revision` field.
+
+- If no traits binding with a component have revision field, OAM runtime will always update workload directly.
+- If `alwaysControl` field is true, OAM runtime will always create a new workload when component revision changed
+  and leave the trait to control two workloads rolling update.
+- If `alwaysControl` field is false, this means the trait will control only if revision path specified. OAM runtime
```

Comment:
Sounds like we can have a trait definition like below?
```yaml
spec:
    revision:
       alwaysControl: false
```

What does this mean?

Reply:
This means you need to check if the trait specify a revisonPath to know whether we need to create new workload with revision name